Kickstarter-iOS/Views/Controllers/PledgeViewController.swift

```diff
@@ -222,13 +253,79 @@ final class PledgeViewController: UIViewController, MessageBannerViewControllerP
     self.paymentMethodsViewController.view.rac.hidden = self.viewModel.outputs.paymentMethodsViewHidden
   }
 
+  private func goToPaymentAuthorization(_ paymentAuthorizationData: PaymentAuthorizationData) {
+    let request = PKPaymentRequest
+      .paymentRequest(
+        for: paymentAuthorizationData.project,
+        reward: paymentAuthorizationData.reward,
+        pledgeAmount: paymentAuthorizationData.pledgeAmount,
+        selectedShippingRule: paymentAuthorizationData.selectedShippingRule,
+        merchantIdentifier: paymentAuthorizationData.merchantIdentifier
+      )
+
+    guard let paymentAuthorizationViewController =
+      PKPaymentAuthorizationViewController(paymentRequest: request) else { return }
+    paymentAuthorizationViewController.delegate = self
+
+    self.present(paymentAuthorizationViewController, animated: true)
+  }
+
+  private func goToThanks(project: Project) {
+    let thanksVC = ThanksViewController.configuredWith(project: project)
+    self.navigationController?.pushViewController(thanksVC, animated: true)
+  }
+
   // MARK: - Actions
 
   @objc private func dismissKeyboard() {
     self.view.endEditing(true)
   }
 }
 
+// MARK: - PKPaymentAuthorizationViewControllerDelegate
+
+extension PledgeViewController: PKPaymentAuthorizationViewControllerDelegate {
+  func paymentAuthorizationViewController(
+    _: PKPaymentAuthorizationViewController,
+    didAuthorizePayment payment: PKPayment,
+    handler completion: @escaping (PKPaymentAuthorizationResult)
+      -> Void
+  ) {
+    let paymentDisplayName = payment.token.paymentMethod.displayName
+    let paymentNetworkName = payment.token.paymentMethod.network?.rawValue
+    let transactionId = payment.token.transactionIdentifier
+
+    self.viewModel.inputs.paymentAuthorizationDidAuthorizePayment(paymentData: (
+      paymentDisplayName,
+      paymentNetworkName,
+      transactionId
+    ))
+
+    STPAPIClient.shared().createToken(with: payment) { [weak self] token, error in
```

Comment:
So we _could_ consider mocking this and adding it to the `Environment` for testing 🤔

Reply:
I think it's not necessary to do that as long as the outputs from the completion block all go directly to the view model so they can be tested. With that in mind, I think I'm actually going to refactor this and send the `token` and `error` from this completion block directly to the view model instead of turning them into an `Either<String, Error?>` type.

It's also nice from the dependency tree perspective to keep all our interactions with 3rd-party APIs at the `Framework` level, which minimizes dependencies in the `Library` framework, theoretically making it easier to extract to a shared library in the future if necessary.

Reply:
Sounds good!